php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -843,18 +843,22 @@ abstract class NDB_BVL_Instrument extends NDB_Page
 
         // Extract the old data and merge it with what was submitted so that we
         // don't overwrite data from other pages.
-        $oldData = $db->pselectOne(
-            "SELECT id.Data FROM flag f
-                    JOIN instrument_data id ON (id.ID=f.DataID)
-                WHERE CommentID=:cid",
-            ['cid' => $this->getCommentID()]
-        );
-
-        // (The PDO driver seems to return null as "null" for JSON column types)
-        if (!empty($oldData) && $oldData !== "null") {
-            $oldData = json_decode($oldData, true);
+        if (!$this->jsonData) {
+            $oldData = $this->getInstanceData() ?? [];
         } else {
-            $oldData = [];
+            $oldData = $db->pselectOne(
```

Comment:
Why does this have an if/else?

getInstanceData should handle both forms of instrument